### ISSUE TITLE
Bridge Django field choices and graphene's enum type

### ISSUE THREAD

**Issue by conradev** [graphql-python/graphene-django#3]
It would be awesome to see something like this:

``` py
class Foo(models.Model):
    TWITTER = 'twitter'
    WEBSITE = 'website'
    FULL_NAME = 'full_name'
    IDENTITY_CHOICES = (
        (TWITTER, 'Twitter'),
        (WEBSITE, 'Website'),
        (FULL_NAME, 'Full Name'),
    )

    main_identity = models.CharField(max_length=10, choices=IDENTITY_CHOICES, default=FULL_NAME)
```

turned into something like this:

``` py
class Identity(graphene.Enum):
    TWITTER = Foo.TWITTER
    WEBSITE = Foo.WEBSITE
    FULL_NAME = Foo.FULL_NAME


class FooType(DjangoObjectType):
    main_identity = graphene.NonNull(Identity)

    class Meta:
        model = Foo
```

automatically. I will work on this if I have time to implement it.

**Comment by yen223** [comment]
The enum class can't have attributes that begin with a digit, but Django choices can.

We'll need to think about how to remap Django choices to conform to GraphQL's `name` spec.

**Comment by morgante** [comment]
I'd love to see this as well.

**Comment by chaffeqa** [comment]
I actaually think this is already in the works, looking at [this bunch of methods](https://github.com/graphql-python/graphene-django/blob/8136223bb1061da92a05031260d356388c2149a7/graphene_django/converter.py#L20)

however I have realized that somehow those are not being invoked even though I am using `DjangoObjectType`.  @syrusakbary  is there someway you can weigh in to what I may be doing wrong? the stacktrace below doesnt seem to involve any reference from `graphene_django` code paths, and I'm having to put in a hacky `resolve` to stop the error (one of the choices is `50/50`):

```
Unhandled exception in thread started by <function check_errors.<locals>.wrapper at 0x10c395620>
Traceback (most recent call last):
  django...
    url(r'^api/', include('api.urls', namespace='api')),
  File "site-packages/django/conf/urls/__init__.py", line 50, in include
    urlconf_module = import_module(urlconf_module)
  File "importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 662, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "src/api/urls.py", line 7, in <module>
    from .schema import schema
  File "src/api/schema.py", line 26, in <module>
    query=Query
  File "site-packages/graphene/types/schema.py", line 38, in __init__
    self.build_typemap()
  File "site-packages/graphene/types/schema.py", line 85, in build_typemap
    self._type_map = TypeMap(initial_types, auto_camelcase=self.auto_camelcase)
  File "site-packages/graphene/types/typemap.py", line 50, in __init__
    super(TypeMap, self).__init__(types)
  File "site-packages/graphql/type/typemap.py", line 16, in __init__
    self.update(reduce(self.reducer, types, OrderedDict()))
  File "site-packages/graphene/types/typemap.py", line 58, in reducer
    return self.graphene_reducer(map, type)
  File "site-packages/graphene/types/typemap.py", line 70, in graphene_reducer
    return self.construct_objecttype(map, type)
  File "site-packages/graphene/types/typemap.py", line 144, in construct_objecttype
    map[type._meta.name]._fields = self.construct_fields_for_type(map, type)
  File "site-packages/graphene/types/typemap.py", line 208, in construct_fields_for_type
    map = self.reducer(map, field.type)
  File "site-packages/graphene/types/typemap.py", line 58, in reducer
    return self.graphene_reducer(map, type)
  File "site-packages/graphene/types/typemap.py", line 70, in graphene_reducer
    return self.construct_objecttype(map, type)
  File "site-packages/graphene/types/typemap.py", line 144, in construct_objecttype
    map[type._meta.name]._fields = self.construct_fields_for_type(map, type)
  File "site-packages/graphene/types/typemap.py", line 208, in construct_fields_for_type
    map = self.reducer(map, field.type)
  File "site-packages/graphene/types/typemap.py", line 58, in reducer
    return self.graphene_reducer(map, type)
  File "site-packages/graphene/types/typemap.py", line 63, in graphene_reducer
    return self.reducer(map, type.of_type)
  File "site-packages/graphene/types/typemap.py", line 58, in reducer
    return self.graphene_reducer(map, type)
  File "site-packages/graphene/types/typemap.py", line 70, in graphene_reducer
    return self.construct_objecttype(map, type)
  File "site-packages/graphene/types/typemap.py", line 144, in construct_objecttype
    map[type._meta.name]._fields = self.construct_fields_for_type(map, type)
  File "site-packages/graphene/types/typemap.py", line 208, in construct_fields_for_type
    map = self.reducer(map, field.type)
  File "site-packages/graphene/types/typemap.py", line 58, in reducer
    return self.graphene_reducer(map, type)
  File "site-packages/graphene/types/typemap.py", line 63, in graphene_reducer
    return self.reducer(map, type.of_type)
  File "site-packages/graphene/types/typemap.py", line 58, in reducer
    return self.graphene_reducer(map, type)
  File "site-packages/graphene/types/typemap.py", line 78, in graphene_reducer
    return self.construct_enum(map, type)
  File "site-packages/graphene/types/typemap.py", line 120, in construct_enum
    description=type._meta.description,
  File "site-packages/graphene/types/definitions.py", line 14, in __init__
    super(GrapheneGraphQLType, self).__init__(*args, **kwargs)
  File "site-packages/graphql/type/definition.py", line 408, in __init__
    self.values = define_enum_values(self, values)
  File "site-packages/graphql/type/definition.py", line 454, in define_enum_values
    assert_valid_name(value_name)
  File "site-packages/graphql/utils/assert_valid_name.py", line 9, in assert_valid_name
    assert COMPILED_NAME_PATTERN.match(name), 'Names must match /{}/ but "{}" does not.'.format(NAME_PATTERN, name)
AssertionError: Names must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but "50_50" does not.
```

**Comment by yen223** [comment]
What we need is some mechanism to override the choice -> Enum mapping for a Django choice field.

**Comment by morgante** [comment]
@chaffeqa It looks like you're running into the issue that enums cannot start with a number (but Django choices can). 3a9ffba56086691360dc68a6d93d3e36e30ea01a fixes that.

**Comment by chaffeqa** [comment]
oooo that explains it... i saw that code but was wondering why it wasnt showing up locally... i think i need to point to pre version of graphene_django?

so im a real python noob :( is there an easy way to do that in a `requirements.txt` file? should i just point to master of github?

**Comment by syrusakbary** [comment]
As @morgante commented this issue is now fixed in master. Closing :)
(will upload a new version to PyPI once the refactor in the connections is done)

**Comment by BossGrand** [comment]
When resolving a choice field, will you have access to both the value and label?

**Comment by spockNinja** [comment]
Cleaning up old resolved PRs.

The basic functionality for this is working in master. If there are any unresolved bugs or specific related features, please open a new issue.
